### ISSUE TITLE
Inference shapes exception with Gemma 2 SPPO

### ISSUE THREAD

**Issue by uogbuji** [ml-explore/mlx-examples#868]
I'll try to simplify the reproduction, but just in case the solution jumps out for anyone, running into this during simple inference:

```pytb
  File "/Users/uche/.local/venv/temp/lib/python3.11/site-packages/mlx_lm/models/gemma2.py", line 176, in __call__
    out = self.model(inputs, cache)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/uche/.local/venv/temp/lib/python3.11/site-packages/mlx_lm/models/gemma2.py", line 159, in __call__
    h = layer(h, mask, c)
        ^^^^^^^^^^^^^^^^^
  File "/Users/uche/.local/venv/temp/lib/python3.11/site-packages/mlx_lm/models/gemma2.py", line 122, in __call__
    r = self.self_attn(self.input_layernorm(x), mask, cache)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/uche/.local/venv/temp/lib/python3.11/site-packages/mlx_lm/models/gemma2.py", line 80, in __call__
    output = mx.fast.scaled_dot_product_attention(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Shapes (1,8,2,32,36) and (32,32) cannot be broadcast.
```

I'm on the latest mlx:

```
pip show mlx mlx_lm 
Name: mlx
Version: 0.15.2
Summary: A framework for machine learning on Apple silicon.
Home-page: https://github.com/ml-explore/mlx
Author: MLX Contributors
Author-email: mlx@group.apple.com
License: 
Location: /Users/uche/.local/venv/temp/lib/python3.11/site-packages
Requires: 
Required-by: mlx-lm, Toolio
---
Name: mlx-lm
Version: 0.15.0
Summary: LLMs on Apple silicon with MLX and the Hugging Face Hub
Home-page: https://github.com/ml-explore/mlx-examples
Author: MLX Contributors
Author-email: mlx@group.apple.com
License: MIT
Location: /Users/uche/.local/venv/temp/lib/python3.11/site-packages
Requires: jinja2, mlx, numpy, protobuf, pyyaml, transformers
Required-by: Toolio
```

**Comment by awni** [comment]
Gemma2 works fine for me, e.g. 

```
python -m mlx_lm.generate --model mlx-community/gemma-2-9b-it-4bit --prompt "Write a story about Einstein" --temp 0.0  --max-tokens 1000
```

runs well.


Could you share the command so we can reproduce it?

**Comment by awni** [comment]
I'm a little confused by your example. You saved to `~/.local/share/models/mlx/Gemma-2-9B-It-SPPO-Iter3-4bit` but loaded from `~/.local/share/models/mlx/Gemma-2-9B-It-SPPO-Iter3-8bit`? 

> I don't know if that looks like something worth the work to accommodate

I'm not sure what needs to be accommodated here.. it looks like a shape mismatch based on possibly using the wrong quantization config. I would double check the config in the ` ~/.local/share/models/mlx/Gemma-2-9B-It-SPPO-Iter3-8bit` path and make sure the `bits=8`.

**Comment by uogbuji** [comment]
OK now is where I confess that I'm all over the place—literally. A user complaint earlier this week led to a hasty investigation in between packing for a trip where I'm on layover as I type. I'll just close this, and thanks for the further nudge in the right direction, which I'll properly digest once I'm properly situated!

**Comment by awni** [comment]
No worries! Let me know if there’s anything here to investigate or feel free to open a new issue if it’s not resolved.
